### ISSUE TITLE
Change neo-hide-cursor bool to a selection for 3+ options

### ISSUE THREAD

**Issue by DivineDominion** [jaypei/emacs-neotree#349]
Hiding the cursor #233 was implemented in #276 

A side-effect of hiding the cursor is that `hl-line-mode` is then enabled automatically.

I wonder if changing the customization option from a boolean to a selection would make sense, with e.g.: `neo-selection-indicator` set to one of `'cursor`, `'hl-line`, `'none` (or nil).

I could make the change, but want to discuss this first.

What do y'all think?

**Comment by DivineDominion** [comment]
Sketched in code:

```
(defcustom neo-selection-indicator 'cursor
  :type '(choice (const :tag "Cursor" 'cursor)
                 (const :tag "None" nil)
                 (const :tag "Highlight Line" 'hl-line)))
```

Instead of constant values set to symbols, might also add a function that's executable, but not sure if I personally prefer that. Interpreting the symbols would be fine with me.

**Comment by Abdisalan** [comment]
Curious, how do you keep track of what line you're on if there's no cursor or highlight?

**Comment by DivineDominion** [comment]
@Abdisalan I don't; this is rather preserving the current customization options (as I understand them). Switching between cursor and line highlight without a 3rd option sounds sensible to me, though.

**Comment by Abdisalan** [comment]
@DivineDominion I agree, we shouldn't be forced to use hl-line if we don't want to. If you put a PR for this change I'll approve it
